minio/minio
# Allow topologySpreadConstraints configuration

## Description

This adds the ability to configure `topologySpreadConstraints` as addition to the existing option for `tolerations`.
It is an alternative to the `affinity` that is not properly capable of allowing multiple pods on one node but also
tries to spread them as much as possible across the cluster.

## Motivation and Context

The default replica value is 16 (right now) which can lead to a massive resource consumption on one node in smaller clusters. The idea for this addition is to allow users to specify how the pods (replicas) are being spread across the cluster. It gives more control over this Helm Release in smaller clusters where most worker nodes have taints.

As this Kubernetes feature exists since Kubernetes 1.19 and is only useful for a replica count > 1, this was taken into account.

## How to test this PR?

- Vary the usage of Kubernetes cluster version 1.18.x and later to see that it is skipped for older version than 1.19
- Mix different values for `.Values.replicas`, `.Values.mode` and `.Values.tolerations`  and use `.Values.topologySpreadConstraints` based on [official docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/)

## Types of changes

## Checklist:
